### PR TITLE
FE Dynamic Modal Component

### PR DESCRIPTION
**Pull request by manda-02**
### **Modal Component**
Modal component complete! I also added DropDown, InputField, and RangeSlider components so it is easier to assemble each modal for every possible action (e.g. delete, confirmations, etc). 

### **Testing**
1. navigate to branch: `git checkout modal`
2. `cd` to the `public` folder
3. run `npm install`
4. run `npm start`
5. ensure the "+ Add" button opens the modal and all text fields are correct and buttons are working for you:)
<br>

When open the modal should look like this:

Screenshot 2024-02-25 at 10 28 13 PM

For reference checkout issue: #10

### CODE REVIEW

File: client/src/components/DropDown.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  label?: string;
+  important?: boolean;
+  type?: string;
+}
+
+const Dropdown: React.FC<DropdownProps> = ({ placeholder = "", label = "", important = false, type = "text" }: DropdownProps) => {
+  const [showDropdown, setShowDropdown] = useState(false);
+  const [selectedOption, setSelectedOption] = useState("");
+  
+  const options = ["Adaptive Functioning", "Option 2", "Option 3"];
```
Comment (farahwamer) [inline client/src/components/DropDown.tsx:15]: I listed out the Possible options for Measure and Item type in the `libraryItem.tsx` file in the models directory, you could use those to list out the options for the dropdowns

File: client/src/components/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                        </div>
+                        <div className='flex flex-row pb-4'>
+                            <div className='pr-4'>
+                                <Dropdown placeholder="C" label="Level of Use" important={false}/>
+                            </div>
+                            <div className='pr-4'>
+                                <InputField placeholder="#  2" label="Edition" important={true} type="Number"/>
+                            </div>
+                            <div>
+                                <InputField placeholder="ABAS-2" label="Acronym" important={true} type="Number"/>
```
Comment (farahwamer) [inline client/src/components/Modal.tsx:62]: Could we make the acronym field a text field, i think Number was a mistake hehe

Comment (manda-02) [reply client/src/components/Modal.tsx:62]: nice catch!

**Review by farahwamer — COMMENTED** [review]
Could we change the font to match the Figma?

**Comment by farahwamer** [comment]
Looks great I just left a few comments! <3

**Review by farahwamer — APPROVED** [review]
looks great!

**Pull request merged**